File: BackEnd/Features/Ships/Registrars/Models/Registrar.cs
using BlueWaterCruises.Features.Ships.Base;
using BlueWaterCruises.Infrastructure.Identity;

namespace BlueWaterCruises.Features.Ships.Registrars {

    public class Registrar {

        // PK
        public int Id { get; set; }
        // Fields
        public string Fullname { get; set; }
        public string Phones { get; set; }
        public string Email { get; set; }
        public string Fax { get; set; }
        public string Address { get; set; }
        public bool IsPrimary { get; set; }
        public bool IsActive { get; set; }
        // FKs
        public int ShipId { get; set; }
        public string UserId { get; set; }
        // Navigation
        public Ship Ship { get; set; }
        public UserExtended User { get; set; }

    }

}

File: FrontEnd/src/app/features/registrars/classes/models/registrar.ts
import { ShipDropdownResource } from './../../../reservations/classes/resources/form/dropdown/ship-dropdown-resource'

export class Registrar {

    id: number
    fullname: string
    phones: string
    email: string
    fax: string
    address: string
    isPrimary: boolean
    isActive: boolean

    ship: ShipDropdownResource

}
